hazelcast/src/main/java/com/hazelcast/nio/serialization/GenericRecord.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.nio.serialization;
+
+import com.hazelcast.internal.serialization.impl.portable.PortableGenericRecordBuilder;
+
+/**
+ * A generic object interface that is returned when the domain class can not be created.
+ * Currently this is valid for {@link Portable} objects.
+ */
+public interface GenericRecord {
+
+    /**
+     * @return an empty generic record builder with same class definition as this one
+     */
+    Builder createGenericRecordBuilder();
```

Comment:
`GenericRecord` in the name seems redundant. just `newBuilder()` ?

Reply:
More javadoc. When this method supposed to be used? How is this different from creating a `Builder` from scratch?